### ISSUE TITLE
Curl: (35) Unknown SSL protocol error in connection to archive.org:443

### ISSUE THREAD

**Issue by harlock74** [theypsilon/Update_All_MiSTer#34]
Hi,
I have got a new Mister FPGA set up. Every time a try to use the mame script I receive this error message:

Curl: (35) Unknown SSL protocol error in connection to archive.org:443

It works perfectly fine with the other Mister FPGA setup. The only difference is the way I connect to internet. In fact with my second setup I use the hotspot function on my phone as the WIFI connection is very poor in the room where the Mister is.

I have also tried to run the security script but to no avail.

Many thanks for looking at my issue!

**Comment by harlock74** [comment]
Hi,
I have got a new Mister FPGA set up. Every time a try to use the script I receive this error message:

`Curl: (35) Unkown SSL protocol error in connection to archive.org:443`

It works perfectly fine with the other Mister FPGA setup. The only difference is the way I connect to internet. In fact with my second I use the hotspot function on my phone as the  WIFI connection is very poor in the room where the Mister is.
Many thanks for looking at my issue!

**Comment by theypsilon** [comment]
Sorry, I totally missed this issue.

Just one question. Have you tried to run the script security_fixes.sh ?

**Comment by harlock74** [comment]
Hi,

Many thanks for your reply. Very much appreciated.

Yes, I have run the scipt.

It's odd as it happens only when I use my smartphone as hot spot to connect
to internet. If I use my wifi router the problem doesn't occur. This
happens with both my Mister setups.

Many thanks for looking into this.

**Comment by theypsilon** [comment]
For the setup using hot spot, I would actually try to remove the changes applied with the security_fixes.sh script. Since SSL connections seem to be a problem, maybe is better to use insecure connections with that setup.

**Comment by harlock74** [comment]
I see many thanks. Can you please let me know how to remove those changes?

Thanks a lot!

**Comment by theypsilon** [comment]
I'm actually not sure about how to do it without starting a setup from scratch.

Other option you have, modify update_mame-getter.ini , update_hbmame-getter.ini , update_mame-getter.ini and update_bios-getter.ini , and in each one of them add the following:
SSL_SECURITY_OPTION="--insecure"

**Comment by harlock74** [comment]
Hi,

Thanks for your recommendations.

I will try those steps. Also I will try with a fresh install with another
sd card to see if it works.

I will keep you posted.

Thanks!

**Comment by harlock74** [comment]
Hi,

I have started from scratch with a fresh install. None of the .ini files
existed in the /media/fat/Scripts so I have created these empty files:

update_mame-getter.ini
update_hbmame-getter.ini

and I have inserted only one line:

SSL_SECURITY_OPTION="--insecure"

But when I run the scrpt I have got the same error message.

Please advise.

Many thanks.

**Comment by theypsilon** [comment]
Could you please share the whole log from running the script? It should be stored at /media/fat/Scripts/.cache/update_all/update_all.log

That way I might find something there that could be useful.

**Comment by harlock74** [comment]
Hi,

Please find attached, as requested.

In order to reproduce my issue, I have removed black tiger .mra and associated rom to force mame-getter to download it again. In fact I have got the usual error message.

By the way I have noticed that if I try to connect with my smartphone to archive.org <http://archive.org/> by using my mobile data connection, instead of my wifi connection, I get the following error message:

This site can’t be reached
The connection was reset
Try:
Checking the connection
ERR_CONNECTION_RESET

Hope that with help for troubleshooting.

Thanks again for looking at it!

**Comment by theypsilon** [comment]
I think the fact that you can't use your phone to access archive.org tells that it's most probably an issue with your phone company. In that case, I'm afraid I can't help you with that. I'm sorry!

**Comment by harlock74** [comment]
Hi,
Thanks for getting back to me.
I have tried with 3 different UK providers on 3 different smartphones and
still the same issue. It's so odd!!
Never mind, many thanks again for looking into this for me. Very much
appreciated!

**Comment by theypsilon** [comment]
Thank you too for reporting the issue. It helps me to keep improving the script :)

**Comment by harlock74** [comment]
Happy days!!
I think I have sorted it out!!!
I have just typed web.archive.org and this time I have got an error message
saying that I had to prove my age (+18) to get access to the website. So I
went to my mobile internet provider home settings and I confirmed my age
and...bingo!!! All working now as expected.

Same scenario with another of my mobile internet providers where the
problem was with the 'data content controls' where I have had to give full
access with no restrictions to any wesites and...bingo again!!!

Happy boy now! Thanks a lot for your help and for giving me the tip that it
was an issue with my internet provider and not with your script which is
spot on. But I am sure this could help others facing the same issue as
mine!

Thanks again!
